sql/validation.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+include "./db-connection.php";
+
+$logfile = fopen("logfile.txt", "w");
+
+$valid = array();
+$errors = array();
+
+define("MAX_LENGTH_SUBJECT", 128);
+define("MAX_LENGTH_LECTURER", 128);
+define("MAX_LENGTH_DESCR", 1024);
+
+if ($_POST) {
+$title = $_POST["title"];
+$lecturer = $_POST["lecturer"];
+$description = $_POST["description"];
+
+if (!$title) {
+$errors["title"] = "Име на предмета е задължително поле.";
+} elseif (strlen($title) > MAX_LENGTH_SUBJECT) {
+$errors["title"] = "Името на предмета има максимална дължина " . MAX_LENGTH_SUBJECT . " символа.";
+} else {
+$valid["title"] = $title;
+}
+
+if (!$lecturer) {
+$errors["lecturer"] = "Преподавател е задължително поле.";
+} elseif (strlen($lecturer) > MAX_LENGTH_LECTURER) {
+$errors["lecturer"] = "Преподавател има максимална дължина " . MAX_LENGTH_LECTURER . " символа.";
+} else {
+$valid["lecturer"] = $lecturer;
+}
```

Comment:
Maybe you can create a field validation function for title, description and lecturer, instead of repeating the same code.